criu/files-reg.c

```diff
@@ -2124,10 +2132,17 @@ int open_path(struct file_desc *d, int (*open_cb)(int mntns_root, struct reg_fil
 	if (orig_path)
 		rfi->path = orig_path;
 
-	if (restore_fown(tmp, rfi->rfe->fown))
+	if (restore_fown(tmp, rfi->rfe->fown)) {
+		close(tmp);
 		return -1;
+	}
 
 	return tmp;
+err:
```

Comment:
nit: I would rename it to err_with_unlock or something like this to make it obvious that it unlocks remap_open_lock

Reply:
Please don't:

1) only unlocks for remap case
2) extra 9 lines change which does not directly related to commit purpose